api/storage_service.cc

```diff
@@ -530,6 +530,48 @@ void set_storage_service(http_context& ctx, routes& r, sharded<service::storage_
         });
     });
 
+    ctx.db.local().get_builtin_routine_registry().register_routine("toppartitions",
+        db::make_single_value_routine<sstring>("toppartitions", [&ctx] (db::builtin_routine::parameter_map&& par) {
+            bool filters_provided = par.contains("table_filters") || par.contains("keyspace_filters");
+
+            std::unordered_set<std::tuple<sstring, sstring>, utils::tuple_hash> table_filters {};
+            if (par.contains("table_filters") && !par["table_filters"].empty()) {
+                std::istringstream ss { std::move(par["table_filters"]) };
```

Comment:
It's hard to evaluate without documentation about what is expected, but how are you supposed to CALL with a multiline string parameter? It's awkward to type.

Should probably be a cql list<text> type, and use the cql type system and parsing, rather than inventing our own here.

Reply:
It's a comma separated list passed as a string (same way as for nodetool and our rest api). This is enforced as of right now because CALL statements only accept strings as parameters.